EIPS/eip-5219.md

```diff
@@ -0,0 +1,75 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-10
+requires: 137, 165, 4804, 4834
+---
+
+## Abstract
+
+Ethereum is the most-established blockchain for building decentralized applications (referred to as `DApp`s). Due to this, the Ethereum DApp ecosystem is very diverse. However, one issue that plagues DApps is the fact that they are not fully decentralized. Specifically, to interface a "decentralized" application, one first needs to access a *centralized* website containing the DApp's front-end code, presenting a few issues (discussed in the Motivation). To solve these issues, this EIP introduces an interface allowing clients to make HTTP-like requests (containing a method, a resource to request, a request body, and headers) to smart contracts, and to receive HTTP-like responses (containing headers and a body).
+
+## Motivation
+
+As mentioned in the Abstract, DApps being partially centralized is problematic. The following are some risks associated with using centralized websites to interface with decentralized applications:
+
+- Trust Minimization: An unnecessarily large number of entities need to be trusted
+- Censorship: A centralized website is not resistant to being censored
+- Permanence: The interface may not have a mechanism that permits it to be permanently stored
+- Interoperability: Smart Contracts cannot directly interact with DApp interfaces
+  
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### Name Resolution
+
+It is RECOMMENDED for clients implementing this EIP to recognize the following standards:
+
+| Name                          | Specification                                        |
+| ---                           | ---                                                  |
+| [EIP-137](./eip-137.md) (ENS) | See [EIP-137](./eip-137.md#resolver-specification)   |
+| [EIP-4834](./eip-4834.md)     | See [EIP-4834](./eip-4834.md#specification)          |
+
+**Changes to this list:** This list MAY be appended to even when this EIP becomes final. New additions should be made at the bottom of the list.
+
+Clients MAY also support regular DNS, as defined in RFC 1034 and RFC 1035.
+
+In the case of a naming conflict, it is RECOMMENDED to prefer the standard that appears first in the list. (Regular DNS should be treated as the last item in the list.)
+
+### Separation of Concerns
+
+It is RECOMMENDED to separate the application logic from the front-end logic (the contract implementing the interface defined in [Contract Interface](#contract-interface)).
+
+### Contract Interface
+
+DApp contracts MUST implement the interface defined in the following file: [Contract Interface](../assets/eip-5219/IDecentralizedApp.sol).
+
+In addition, DApp contracts MUST implement [EIP-165](./eip-165.md).
```

Comment:
Why is this required?  It doesn't feel like it is strictly necessary for this to be a useful specification, and one can imagine a future mechanism for interface detection that is superior that the ecosystem wants to switch to, and it would be unfortunate for anything using that newer/better system to no longer be compatible with this EIP.